sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -180,3 +182,214 @@ export function getAssociatedReceiverName(
+ * @param value
+ */
+export function getJSONOrUndefined(value: any): number | undefined {
+  if (value == undefined || (typeof value === "string" && !value.toString())) {
+    return undefined;
+  } else {
+    try {
+      return JSON.parse(JSON.stringify(value));
+    } catch (err) {
+      throw new TypeError(
+        `${JSON.stringify(value, undefined, 2)} expected to be in proper JSON format, or undefined`
+      );
+    }
+  }
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve message count details from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getCountDetailsOrUndefined(value: any): MessageCountDetails | undefined {
+  const jsonValue: any = getJSONOrUndefined(value);
+  if (jsonValue != undefined) {
+    try {
+      const result: MessageCountDetails = {
+        activeMessageCount: parseInt(jsonValue["d2p1:ActiveMessageCount"]),
+        deadLetterMessageCount: parseInt(jsonValue["d2p1:DeadLetterMessageCount"]),
+        scheduledMessageCount: parseInt(jsonValue["d2p1:ScheduledMessageCount"]),
+        transferMessageCount: parseInt(jsonValue["d2p1:TransferMessageCount"]),
+        transferDeadLetterMessageCount: parseInt(jsonValue["d2p1:TransferDeadLetterMessageCount"])
+      };
+      return result;
+    } catch (err) {
+      throw new TypeError(
+        `${jsonValue} expected to be in expected CountDetails format with numbers, or undefined`
+      );
+    }
+  } else {
+    return undefined;
+  }
+}
+
+/**
+ * Represents type of message count details in ATOM based management operations
+ */
+export type MessageCountDetails = {
+  activeMessageCount: number;
+  deadLetterMessageCount: number;
+  scheduledMessageCount: number;
+  transferMessageCount: number;
+  transferDeadLetterMessageCount: number;
+};
+
+/**
+ * Represents type of `AuthorizationRule` in ATOM based management operations
+ */
+export type AuthorizationRule = {
+  claimType: string;
+  claimValue: string;
+  rights: { accessRights?: string[] };
+  keyName: string;
+  primaryKey?: string;
+  secondaryKey?: string;
+};
+
+/**
+ *  @ignore
+ * Helper utility to retrieve array of `AuthorizationRule` from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getAuthorizationRulesOrUndefined(value: any): AuthorizationRule[] | undefined {
+  const authorizationRules: AuthorizationRule[] = [];
+  const jsonValue: any = getJSONOrUndefined(value);
+  if (jsonValue == undefined) {
+    return undefined;
+  } else {
+    try {
+      const rawAuthorizationRules = value["AuthorizationRule"];
+      if (rawAuthorizationRules.length !== undefined) {
+        for (let i = 0; i < rawAuthorizationRules.length; i++) {
+          authorizationRules.push(buildAuthorizationRule(rawAuthorizationRules[i]));
+        }
+      } else {
+        authorizationRules.push(buildAuthorizationRule(rawAuthorizationRules));
+      }
+      return authorizationRules;
+    } catch (err) {
+      throw new TypeError(
```

Comment:
This is different from the feature agnostic utility in that it's more related to serialization and is the caller for the JSON utility.
I'd keep this check as it specifically wraps service specific serialization code.
Also because it helps catch errors faster that might otherwise result in issues that get masked elsewhere.